### ISSUE TITLE
EDG (edg-6_8-default-15) silently emits no diagnostics when invoked with -S

### ISSUE THREAD

**Issue by Fedr** [compiler-explorer/compiler-explorer#8686]
The EDG front-end as configured on Compiler Explorer is invoked as

```
edgcpfe -g -o output.s -S --c++23 example.cpp
```

for the normal compile pane. With `-S`, `edgcpfe` exits 0 with empty stderr and produces a placeholder `<No output file>` — even when the program contains a hard error such as a failing `static_assert`. Toggling **Execute the code** triggers a second invocation without `-S`, which does report the error correctly.

### Repro

https://gcc.godbolt.org/z/M4WMPcoz3 — same source, two EDG `edg-6_8-default-15` panes with `--c++23`. Left pane has execute off (no diagnostic, empty output). Right pane has execute on (the static_assert error appears).

Source:

```cpp
template<int N>
struct s{
  constexpr static       int&& p = N;
};

template<typename T> void g() {
    static_assert(T::p == 1);
}

int main(){
   g<s<1>>();
}
```

With execute on, EDG correctly reports:

```
"<source>", line 7: error: expression must have a constant value
      static_assert(T::p == 1);
                    ^
"<source>", line 7: note: attempt to access run-time storage
      static_assert(T::p == 1);
                    ^
          detected during instantiation of "void g<T>() [with T=s<1>]"
```

### Expected

Front-end diagnostics (errors, `static_assert` failures, template-instantiation errors) should be visible from the standard compile view, not gated on enabling execution.

### Likely cause

The `-S` mode for this EDG entry seems to skip full semantic analysis / template instantiation. Either drop `-S` for the EDG entry or pair it with a flag that still forces full analysis.

The issue was originally found here: https://stackoverflow.com/q/79937637/7325599
